PurchasesTests/Support/BeginRefundRequestHelperTests.swift

```diff
@@ -33,126 +98,232 @@ class BeginRefundRequestHelperTests: XCTestCase {
+        identityManager = MockIdentityManager(mockAppUserID: "appUserID")
+        helper = BeginRefundRequestHelper(systemInfo: systemInfo,
+                                          customerInfoManager: customerInfoManager,
+                                          identityManager: identityManager)
 
         if #available(iOS 15.0, macCatalyst 15.0, *) {
             helper.sk2Helper = sk2Helper
         }
+
+    }
+
+    func testBeginRefundRequestForProductFatalErrorIfNotIosOrCatalyst() throws {
+        #if os(watchOS) || os(macOS) || os(tvOS)
+        expectFatalError(expectedMessage: Strings.purchase.begin_refund_request_unsupported.description) {
+            _ = try await helper.beginRefundRequest(forProduct: mockProductID)
+        }
+        #else
+        throw XCTSkip("beginRefundRequestForProduct is not available for this test.")
+        #endif
     }
 
-    func testBeginRefundRequestFatalErrorIfNotIosOrCatalyst() {
+    func testBeginRefundRequestForEntitlementFatalErrorIfNotIosOrCatalyst() throws {
         #if os(watchOS) || os(macOS) || os(tvOS)
         expectFatalError(expectedMessage: Strings.purchase.begin_refund_request_unsupported.description) {
-            helper.beginRefundRequest(productID: mockProductID) { result in }
+            _ = try await helper.beginRefundRequest(forEntitlement: mockEntitlementID)
         }
+        #else
+        throw XCTSkip("beginRefundRequestForEntitlement is not available for this test.")
         #endif
     }
 
-#if os(iOS) || targetEnvironment(macCatalyst)
-    func testBeginRefundRequestFailsAndPassesErrorThroughIfPurchasesUnverified() throws {
+    func testBeginRefundRequestForActiveEntitlementFatalErrorIfNotIosOrCatalyst() throws {
+        #if os(watchOS) || os(macOS) || os(tvOS)
+        expectFatalError(expectedMessage: Strings.purchase.begin_refund_request_unsupported.description) {
+            _ = try await helper.beginRefundRequestforActiveEntitlement()
+        }
+        #else
+        throw XCTSkip("beginRefundRequestForEntitlement is not available for this test.")
+        #endif
+    }
+
+#if os(iOS)
+    func testBeginRefundRequestFailsAndPassesErrorThroughIfPurchasesUnverified() async throws {
         guard #available(iOS 15.0, macCatalyst 15.0, *) else {
             throw XCTSkip("Required API is not available for this test.")
         }
-
         let expectedError = ErrorUtils.beginRefundRequestError(withMessage: "test")
 
         sk2Helper.transactionVerified = false
         sk2Helper.maybeMockSK2Error = expectedError
 
+        do {
+            _ = try await helper.beginRefundRequest(forProduct: mockProductID)
+            XCTFail("beginRefundRequestForProduct should have thrown error")
+        } catch {
+            expect(self.sk2Helper.verifyTransactionCalled).to(beTrue())
+            // confirm we don't call refund request method if transaction not verified
+            expect(self.sk2Helper.refundRequestCalled).to(beFalse())
```

Comment:
I believe this block could also be simplified as

```swift
expect { _ = try await helper.beginRefundRequest(forProduct: mockProductID) }.to(throwError)

expect(self.sk2Helper.verifyTransactionCalled).to(beTrue())
// confirm we don't call refund request method if transaction not verified
expect(self.sk2Helper.refundRequestCalled).to(beFalse())
```

without the need of the `do / try / catch` bit

I haven't tried it with `async / await` yet, though, so FFTI if it's not compatible.

Reply:
just tried it and it's not compatible 😢 so if you try my suggestion it'll error out saying that no errors were thrown. Disregard

Reply:
good to know for other cases though, thanks!